cluster/photon-controller/setup-prereq.sh

```diff
@@ -118,40 +141,95 @@ function create-vm-flavor {
     local flavor_name=$1
     local flavor_spec=$2
     local rc=0
+    local output
 
     $PHOTON flavor list | grep "\t${flavor_name}\t" > /dev/null 2>&1 || rc=$?
     if [ $rc -eq 0 ]; then
+        check-flavor-ready ${flavor_name}
         echo "Flavor ${flavor_name} already made, skipping"
     else
         echo "Making VM flavor ${flavor_name}"
-        $PHOTON -n flavor create --name "${flavor_name}" --kind "vm" --cost "${flavor_spec}" > /dev/null 2>&1
+        rc=0
+        output=$($PHOTON -n flavor create --name "${flavor_name}" --kind "vm" --cost "${flavor_spec}" 2>&1) || rc=$?
+        if [ $rc -ne 0 ]; then
+            echo "ERROR: Could not create vm flavor \"${flavor_name}\", exiting"
+            echo "Output from flavor creation:"
+            echo ${output}
+            exit 1
+        fi
     fi
 }
 
 function create-disk-flavor {
     local rc=0
+    local output
 
     $PHOTON flavor list | grep "\t${PHOTON_DISK_FLAVOR}\t" > /dev/null 2>&1  || rc=$?
     if [ $rc -eq 0 ]; then
+        check-flavor-ready ${PHOTON_DISK_FLAVOR}
         echo "Flavor ${PHOTON_DISK_FLAVOR} already made, skipping"
     else
         echo "Making disk flavor ${PHOTON_DISK_FLAVOR}"
-        $PHOTON -n flavor create --name ${PHOTON_DISK_FLAVOR} --kind "ephemeral-disk" --cost "${SETUP_DISK_FLAVOR_SPEC}" > /dev/null 2>&1
+        rc=0
+        output=$($PHOTON -n flavor create --name ${PHOTON_DISK_FLAVOR} --kind "ephemeral-disk" --cost "${SETUP_DISK_FLAVOR_SPEC}" 2>&1) || rc=$?
+        if [ $rc -ne 0 ]; then
+            echo "ERROR: Could not create disk flavor \"${PHOTON_DISK_FLAVOR}\", exiting"
+            echo "Output from flavor creation:"
+            echo ${output}
+            exit 1
+        fi
+    fi
+}
+
+function check-flavor-ready {
```

Comment:
You could have passed in options for flavor list name and kind rather than making two requests. Thoughts?